backend/pkg/contenttype/middleware.go

```diff
@@ -0,0 +1,91 @@
+package contenttype
+
+import (
+	"errors"
+	"fmt"
+	"mime"
+	"net/http"
+	"strings"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/mendersoftware/mender-server/pkg/rest.utils"
+)
+
+type MiddlewareOption struct {
+	ContentType *string
+	Path        string
```

Comment:
Filtering by path/method shouldn't be necessary as the middleware should only be applied to the routes that use it when setting up the routes.